test/metabase/models/dashboard_test.clj

```diff
@@ -29,168 +30,212 @@
-   (is (= "moved this Dashboard to New collection."
+             {:cards [{:id 1} {:id 2}]}
+             {:cards [{:id 1} {:id 3}]})))))
+
+ (testing "update collection ---"
+  (t2.with-temp/with-temp
+    [Collection {coll-id :id} {:name "New collection"}]
+    (is (= "moved this Dashboard to New collection."
+           (build-sentence
+             (revision/diff-strings
+               Dashboard
+               {:name "Apple"}
+               {:name          "Apple"
+                :collection_id coll-id})))))
+  (t2.with-temp/with-temp
+    [Collection {coll-id-1 :id} {:name "Old collection"}
+     Collection {coll-id-2 :id} {:name "New collection"}]
+    (is (= "moved this Dashboard from Old collection to New collection."
+           (build-sentence
+             (revision/diff-strings
+               Dashboard
+               {:name          "Apple"
+                :collection_id coll-id-1}
+               {:name          "Apple"
+                :collection_id coll-id-2}))))))
+
+ (testing "update tabs"
+   (is (= "added a tab."
           (build-sentence
             (revision/diff-strings
               Dashboard
-              {:name "Apple"}
-              {:name          "Apple"
-               :collection_id coll-id})))))
- (t2.with-temp/with-temp
-   [Collection {coll-id-1 :id} {:name "Old collection"}
-    Collection {coll-id-2 :id} {:name "New collection"}]
-   (is (= "moved this Dashboard from Old collection to New collection."
+              {:tabs [{:id 0 :name "First tab" :position 0}]}
+              {:tabs [{:id 0 :name "First tab" :position 0}
+                      {:id 1 :name "Second tab" :position 1}]}))))
+
+   (is (= "removed 2 tabs."
           (build-sentence
             (revision/diff-strings
               Dashboard
-              {:name          "Apple"
-               :collection_id coll-id-1}
-              {:name          "Apple"
-               :collection_id coll-id-2}))))))
+              {:tabs [{:id 0 :name "First tab" :position 0}
+                      {:id 1 :name "Second tab" :position 1}
+                      {:id 2 :name "Third tab" :position 2}]}
+              {:tabs [{:id 0 :name "First tab" :position 0}]}))))
 
+   (is (= "rearranged the tabs."
+          (build-sentence
+            (revision/diff-strings
+              Dashboard
+              {:tabs [{:id 0 :name "First tab" :position 0}
+                      {:id 1 :name "Second tab" :position 1}]}
+              {:tabs [{:id 1 :name "First tab" :position 0}
+                      {:id 0 :name "Second tab" :position 1}]}))))
+   (is (= "modified the tabs."
+          (build-sentence
+            (revision/diff-strings
+              Dashboard
+              {:tabs [{:id 0 :name "First tab" :position 0}
+                      {:id 1 :name "Second tab" :position 1}]}
+              {:tabs [{:id 1 :name "First tab new name" :position 0}
+                      {:id 0 :name "Second tab new name" :position 1}]}))))))
```

Comment:
The point of this test is the tab IDs changed position and the name changed, but it's not super clear. This might make it clearer:
```suggestion
              {:tabs [{:id 0 :name "tab A" :position 0}
                      {:id 1 :name "tab B" :position 1}]}
              {:tabs [{:id 1 :name "tab B new name and position" :position 0}
                      {:id 0 :name "tab A new name and position" :position 1}]}))))))
```

Reply:
updated in c7a6e58f97